numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -2708,6 +2710,80 @@ npyiter_allocate_arrays(NpyIter *iter,
         bufferdata = NIT_BUFFERDATA(iter);
     }
 
+    if (flags & NPY_ITER_COPY_IF_OVERLAP) {
+        /* Perform operand memory overlap checks if requested */
```

Comment:
Could add more detail on what is being checked here, the notes below are a bit too local.